### PR TITLE
Allow servers to set appeal links, which would be sent with every ban message

### PR DESCRIPTION
**Pull request by RealSGII2**
Users will commonly rejoin servers to ask for ban links, which could be seen as ban bypassing and ruin their chance to appeal.

This PR's goal is to allow server admins to set an appeal link, which would be sent along with the reason in every ban message, This would remove the need to bypass a ban.
image
image
image

### CODE REVIEW

**Comment by Blackcatmaxy** [comment]
Probably want to appeal link in separate embed field both for visibility and less length errors, but otherwise looks good

**Comment by RealSGII2** [comment]
There are a couple issues I have with your suggestion:
1. Adding a separate embed adds clutter and looks bad.
2. This sort of falls into patching the problem rather than preventing it.

A more feasible solution would be to add a separate field, and add more text (i.e. "If you'd like to appeal your ban, _click here_.") there. We could also make the link bold to make it stand out more.

**Comment by Blackcatmaxy** [comment]
A separate field is exactly what I meant

**Comment by RealSGII2** [comment]
Oops, I only saw "embed". Sorry.

**Comment by RealSGII2** [comment]
An additional, small change will be made in this pull request:
- Permanent ban messages will now be in the embed format, replacing the old text style.

This is the old message:
image

And this is the new one:
image

**Comment by Blackcatmaxy** [comment]
Might want to have a conversation about how we want to format links for consistency, so far at least in logs we've included the "click" in "click here" for the links. Probably makes it easier to press on mobile too.
image

**Comment by RealSGII2** [comment]
Comment from the Discord:
> we should probably be consistent with "set command" behavior and accept null/none as valid "reset" inputs

File: src/BotCatMaxy/Components/Moderation/ModerationCommands.cs [hunk 1]
```diff
@@ -174,8 +176,10 @@ public async Task<RuntimeResult> Ban([RequireHierarchy] UserRef userRef, [Remain
             {
                 return CommandResult.FromError("User has already been banned permanently.");
             }
+
             if (userRef.User != null)
-                await userRef.User.TryNotify($"You have been perm banned in the {Context.Guild.Name} discord for {reason}");
+                await userRef.User.Notify($"permanently banned", reason, Context.Guild, Context.Message.Author, appealLink: settings.appealLink);
+                // await userRef.User.TryNotify($"You have been perm banned in the {Context.Guild.Name} discord for {reason}");
```
Comment (Blackcatmaxy) [inline src/BotCatMaxy/Components/Moderation/ModerationCommands.cs]: Probably want to delete this comment

File: src/BotCatMaxy/Utilities/NotifyUtilities.cs [hunk 2]
```diff
@@ -13,14 +13,20 @@ public static class NotifyUtilities
         /// <summary>
         /// Embeds a DM to a user about an action with a reason
         /// </summary>
-        public static async Task Notify(this IUser user, string action, string reason, IGuild guild, IUser author = null, Color color = default)
+        public static async Task Notify(this IUser user, string action, string reason, IGuild guild, IUser author = null, Color color = default, string appealLink = "")
```
Comment (Blackcatmaxy) [inline src/BotCatMaxy/Utilities/NotifyUtilities.cs:16]: probably better to set appealLink default to `null`

File: src/BotCatMaxy/Components/Moderation/Settings.cs [hunk 3]
```diff
@@ -45,6 +46,27 @@ public async Task<RuntimeResult> ToggleServerIDUse()
...
+            ModerationSettings settings = Context.Guild.LoadFromFile<ModerationSettings>(true);
+
+            if (link.IsNullOrEmpty())
+                settings.appealLink = null;
+            else
+                settings.appealLink = link;
+
+            settings.SaveToFile();
+
+            if (link.IsNullOrEmpty() || link == "none" || link == "null")
```
Comment (Blackcatmaxy) [inline src/BotCatMaxy/Components/Moderation/Settings.cs]: Want to use .`ToLower()` so it doesn't complain when a user randomly capitalizes it (look for bad example in `!setpubchannel` in `LoggingCommands`

**Review by Blackcatmaxy — CHANGES_REQUESTED** [review]

File: src/BotCatMaxy/Components/Moderation/ModerationCommands.cs (hunk 1 above)
Comment (RealSGII2) [reply src/BotCatMaxy/Components/Moderation/ModerationCommands.cs]: Oops! 😅

File: src/BotCatMaxy/Components/Moderation/Settings.cs (hunk 3 above)
Comment (RealSGII2) [reply src/BotCatMaxy/Components/Moderation/Settings.cs]: Should I change it for the other settings? (I looked at your older code for reference whether I should do that or not.)

Comment (Blackcatmaxy) [reply src/BotCatMaxy/Components/Moderation/Settings.cs]: We should keep this PR consistent with what we can and do another pass later

**Review by Blackcatmaxy — APPROVED** [review]

**Pull request merged**